### ISSUE TITLE
AutoSeoUrls: Failed, pages not accessible using SEO friendly URLs 

### ISSUE THREAD

**Issue by mxmilkiib** [Jemt/SitemagicCMS#152]
I'm seeing this in the log and realising now why URLs haven't been clean.

**Comment by Jemt** [comment]
@mxmilkiib Please provide the full log output.

**Comment by Jemt** [comment]
@mxmilkiib Please provide the full content of SMLog.xml.php

**Comment by Jemt** [comment]
Sitemagic CMS was not able to automatically enable SEO friendly URLs due to this problem:
file_get_contents(https://xxxxxx.xyz/index.html): failed to open stream: HTTP request failed! HTTP/1.1 404 Not Found

Your server is not able to resolve your website using file_get_contents(..) - either due to strict configuration of misconfiguration, or perhaps due to lack of support for URL rewriting.

Can you access the front page of your website using /index.html ?

**Comment by Jemt** [comment]
If your pages are accessible using `<Filename>.html`, then you can simply enable SEO friendly URLs manually by adding the following entries to /config.xml.php:

```
<entry key="SMPagesSettingsUrlType" value="Filename" />
<entry key="SMPagesSettingsSeoUrls" value="true" />
```

Existing links won't be updated - you will need to update them manually through _Content > Menu_ and _Content > Pages_.

**Comment by mxmilkiib** [comment]
Aha! Right, so I missed the settings on Pages for SEO URLs and URL type. I guess maybe I was looking in settings before, or assuming it would just be on. I clocked reference to first in the log.

I've now turned on SEO URLs and flipped the URL type to filename, and I have edited file properties, and gone in and resaved the file for goot measure, but (I tried editing existing menu links first but got confused) when I try to add a new menu link, I enter a title then Select the page, then it puts "pagefilename.html" in the URL box, I hit Create, but the new link gives a 404.

Possibly the Yunohost Nginx configuration isn't correct..

*looks*

Aah, I see there is an htaccess file, but no reference to Nginx, which is the system [Yunohost ships with](https://yunohost.org/ru/web_server) (which, to reference the old meme, [Netcraft confirms](https://www.netcraft.com/blog/june-2023-web-server-survey/) is the most used server currently).

The Yunohost site provision for SitemagicCMS is done by https://github.com/YunoHost-Apps/SitemagicCMS_ynh which was created in 2021 and looks to be actively developed.

I can ask there about mitigating the issue relating to using a "non standard" web server when it comes to functioning clean URLs.
